hypervisor/boot/sbl/multiboot.c

```diff
@@ -132,43 +159,71 @@ static void *get_kernel_load_addr(void *kernel_src_addr)
  *    vm            pointer to vm structure
  *    cmdline       pointer to cmdline string
  *
+ * output:
+ *    index         pointer to argument index
+ *
  * return value:
- *    boot_params   HVA of image_boot_params if parse success, or
+ *    param_addr    HVA of image_boot_params if parse success, or
  *                  else a NULL pointer.
  */
-static void *parse_image_boot_params(struct vm *vm, char *cmdline)
+static void *parse_image_boot_params(struct vm *vm, char *cmdline, uint32_t *index)
```

Comment:
for the "index", suggest to rename as "bootloader", and define 3 enum (ABL, SBL, unknown) for it.